### ISSUE TITLE
v0.15 introduce naming conflicts in Generated Marshaling Code which will fail the application code compile

### ISSUE THREAD

**Issue by praenubilus** [openenclave/openenclave#3982]
The definition in EDL
```c
untrusted {
        void my_method(
            [string, in] const char* key_name,
            [string, in] const char* key_vault_app_url,
            [string, in] const char* key_vault_url,
            [string, in] const char* report_base64url,
            [string, in] const char* enclave_private_data_base64url,
            [count=32, in] uint8_t *mrsigner, 
            size_t mrsigner_size,
            [out] int* result);
```
The snippet in generated marshaling code `my_method_t.c`

```c
typedef struct _my_method_args_t
{
    oe_result_t result;
    uint8_t* deepcopy_out_buffer;
    size_t deepcopy_out_buffer_size;
    char* key_name;
    size_t key_name_len;
    char* key_vault_app_url;
    size_t key_vault_app_url_len;
    char* key_vault_url;
    size_t key_vault_url_len;
    char* report_base64url;
    size_t report_base64url_len;
    char* enclave_private_data_base64url;
    size_t enclave_private_data_base64url_len;
    uint8_t* mrsigner;
    size_t mrsigner_size;
    int* result;
} secure_key_create_args_t;
```
`oe_result_t result` and `int* result` conflict with each other.

**Comment by mingweishih** [comment]
Yes, we used to declare
`oe_result_t _result;`
Was changed to `result` to conform the [OE naming convention]( https://github.com/openenclave/openenclave/blob/master/docs/Community/DevelopmentGuide.md#naming-conventions).

To solve this, we can add a prefix to the variable (e.g., `oe_result`)

**Comment by praenubilus** [comment]
Just checked today, the fix https://github.com/openenclave/oeedger8r-cpp/pull/78 has not been included in the v0.16 release. Is it correct @radhikaj ?

**Comment by mingweishih** [comment]
@manojrupireddy can you update the oeedger8r submodule?

**Comment by manojrupireddy** [comment]
Yes Ming-wei I will do it.

**Comment by rs--** [comment]
@manojrupireddy is this resolved by #4044 ?

**Comment by manojrupireddy** [comment]
@rs--  yes
